sphinx/themes/basic/static/searchtools.js

```diff
@@ -97,6 +97,14 @@ const _displayItem = (item, searchTerms, highlightTerms) => {
     fetch(requestUrl)
       .then((responseData) => responseData.text())
       .then((data) => {
+
+        // exclude results that don't contain exact phrases if we are searching for them
+        if (data) {
+          const lowercaseData = data.toLowerCase();
+          const mismatch = (s) => !lowercaseData.includes(s);
```

Comment:
Suggestion: perhaps we could/should add word boundaries around the match?

```suggestion
          const mismatch = (s) => !s.match(`\b${lowercaseData}\b`);
```

Reasoning:
  * Could make it easier to exact-search for strings that are substrings of other phrases/words.
  * Although regex usage can introduce some overhead, there's also optimization opportunity if the matching can skip over non-word boundary match positions.